modules/020-deckhouse/hooks/generate_webhook_tls_certificate_test.go

```diff
@@ -109,9 +105,108 @@ var _ = Describe("Deckhouse hooks :: generate_selfsigned_ca ::", func() {
 
 		It("Cert data must be stored in values", func() {
 			Expect(f).To(ExecuteSuccessfully())
-			Expect(f.ValuesGet("deckhouse.internal.webhookHandlerCert.crt").String()).To(Equal("a\n"))
-			Expect(f.ValuesGet("deckhouse.internal.webhookHandlerCert.key").String()).To(Equal("b\n"))
-			Expect(f.ValuesGet("deckhouse.internal.webhookHandlerCert.ca").String()).To(Equal("c\n"))
+			Expect(f.ValuesGet("deckhouse.internal.webhookHandlerCert.crt").String()).To(Equal(actualCert))
+			Expect(f.ValuesGet("deckhouse.internal.webhookHandlerCert.key").String()).To(Equal(actualKey))
+			Expect(f.ValuesGet("deckhouse.internal.webhookHandlerCert.ca").String()).To(Equal(actualCA))
+		})
+	})
+
+	Context("Expired certificate", func() {
+		BeforeEach(func() {
+			f.KubeStateSet(expiredSecret)
+			f.BindingContexts.Set(f.GenerateBeforeHelmContext())
+			f.RunHook()
+		})
+
+		It("Cert data must be changed", func() {
+			Expect(f).To(ExecuteSuccessfully())
+			Expect(f.ValuesGet("deckhouse.internal.webhookHandlerCert.crt").String()).ToNot(Equal(expiredCert))
+			Expect(f.ValuesGet("deckhouse.internal.webhookHandlerCert.key").String()).ToNot(Equal(expiredKey))
+			Expect(f.ValuesGet("deckhouse.internal.webhookHandlerCert.ca").String()).ToNot(Equal(expiredCA))
 		})
 	})
 })
+
+var (
+	actualCert = ""
+	actualCA   = ""
+	actualKey  = ""
+
+	expiredCert = ""
+	expiredCA   = ""
+	expiredKey  = ""
```

Comment:
Just... why? Switching global values from generators seems so indirect. You could use returned values instead and their fields.

Reply:
Well, I have changed it but in my opinion it looks more dirty, too many return parameters. But I don't care too much about tests structure, so, it's up to you

Reply:
Thank you!